examples/go/stream_benchmarker/benchmark/main.go

```diff
@@ -0,0 +1,261 @@
+// Copyright © 2019 Infostellar, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	stellarstation "github.com/infostellarinc/go-stellarstation/api/v1"
+	"github.com/infostellarinc/stellarstation-api/examples/go/benchmark"
+)
+
+type metric struct {
+	timeFirstByteReceived time.Time
+	timeLastByteReceived  time.Time
+	dataSize              int
+	metricTime            time.Time
+}
+
+type outputDetails struct {
+	timenow                        time.Time
+	mostRecentTimeLastByteReceived time.Time
+	averageFirstByteLatency        time.Duration
+	averageLastByteLatency         time.Duration
+	totalDataSize                  int64
+	metricsCount                   int64
+	averageDataSize                float64
+	mbps                           float64
+	outOfOrderData                 int64
+}
+
+type sessionSummary struct {
+	firstByteTime                time.Time
+	lastByteTime                 time.Time
+	totalAverageFirstByteLatency time.Duration
```

Comment:
I named it like this because there is some extra logic in the print function:
(totalAverageFirstByteLatency / numIntervals) to get the average for that pass.